dali/pipeline/operator/common.h

```diff
@@ -38,5 +42,116 @@ inline void GetSingleOrRepeatedArg(const OpSpec &spec, vector<T> &result,
       to_string(result.size()) + " given.");
 }
 
+template <typename T>
+void GetPerSampleArgument(std::vector<T> &output,
+                          const std::string &argument_name,
+                          const OpSpec &spec,
+                          const ArgumentWorkspace &ws,
+                          int batch_size = -1 /* -1 = get from "batch_size" arg */) {
+  if (batch_size < 0)
+    batch_size = spec.GetArgument<int>("batch_size");
+
+  if (spec.HasTensorArgument(argument_name)) {
+    const auto &arg = ws.ArgumentInput(argument_name);
+    decltype(auto) shape = arg.shape();
+    int N = shape.num_samples();
+    if (N == 1) {
+      bool is_valid_shape = shape.tensor_shape(0) == TensorShape<1>{batch_size};
+
+      DALI_ENFORCE(is_valid_shape,
+        make_string("`", argument_name, "` must be a 1xN or Nx1 (N = ", batch_size,
+                    ") tensor list. Got: ", shape));
+
+      output.resize(batch_size);
+      auto *data = arg[0].template data<T>();
+
+      for (int i = 0; i < batch_size; i++) {
+        output[i] = data[i];
+      }
```

Comment:
std::copy?

Reply:
Could be, but I find explicit loop more readable - at least until we have C++ ranges.